src/targets/maven.ts

```diff
@@ -0,0 +1,389 @@
+    );
+  }
+
+  /**
+   * Retrieves a record of all the required files by Maven CLI to upload
+   * anything.
+   *
+   * @param distDir directory of the distribution.
+   * @returns record of required files.
+   */
+  private getFilesForMavenCli(distDir: string): Record<string, string> {
+    const moduleName = parse(distDir).base;
+    const targetFile = join(distDir, this.getTargetFilename(distDir));
+    const javadocFile = join(distDir, `${moduleName}-javadoc.jar`);
+    const sourcesFile = join(distDir, `${moduleName}-sources.jar`);
+    const pomFile = join(distDir, 'pom-default.xml');
+    return {
+      targetFile,
+      javadocFile,
+      sourcesFile,
+      pomFile,
+    };
+  }
+
+  /**
+   * Retrieves the target file name for the current distribution.
+   *
+   * If the distibution is an Android distribution, the target file is the
+   * file containing "release" in the name and the ".aar" extension.
+   * Typically, the module (directory) name without the version and appending
+   * "-release.aar" at the end.
+   *
+   * If the distribution isn't an Android distribution, the target filename is
+   * the module name appending ".jar" to the end.
+   *
+   * @param distDir directory where distributions are.
+   * @returns the target file name.
+   */
+  private getTargetFilename(distDir: string): string {
+    const moduleName = parse(distDir).base;
+    const isAndroidDistDir = this.config.androidDistDirPattern.test(moduleName);
+    if (isAndroidDistDir) {
+      return moduleName.replace(
+        this.config.androidFileSearchPattern,
+        this.config.androidFileReplaceStr
+      );
+    }
+    return `${moduleName}.jar`;
+  }
+
+  /**
+   * Retries executing the function, delaying ongoing calls for some time.
+   *
+   * The function is retried `MAX_PUBLISHING_ATTEMPTS` times.
+   * The delay after each call starts at `RETRY_DELAY_SECS`, and is incremented
+   * exponentially by `RETRY_EXP_FACTOR`.
+   *
+   * @param processFn function to be retried.
+   * @param actionName name of the action the function is performing.
+   */
+  private async retrySpawnProcess(
+    processFn: () => Promise<any>,
+    actionName: string
+  ): Promise<void> {
+    let retryDelay = RETRY_DELAY_SECS;
+    await withRetry(
+      () => processFn(),
+      MAX_PUBLISHING_ATTEMPTS,
+      async err => {
+        this.logger.warn(
+          `${actionName} failed. Trying again in ${retryDelay}s.`
+        );
+        this.logger.debug(`${actionName} error: ${err}`);
+        await sleep(retryDelay * 1000);
+        retryDelay *= RETRY_EXP_FACTOR;
+        return true;
+      }
+    );
+  }
+
+  /**
+   * Finishes the release flow, by closing and releasing to the repository.
+   * Note that this is a required step to make a release, but it doesn't
+   * perform any release on its own. Required files must have previously been
+   * uploaded accordingly.
+   */
+  public async closeAndRelease(): Promise<void> {
+    this.retrySpawnProcess(
```

Comment:
Why are we retrying this one? Comments around why we do retries would be very useful.